### ISSUE TITLE
Symbolic representation of an undefined value

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/ModelicaSpecification#252]
**Reported by mtiller on 5 Jan 2010 14:28 UTC**
In Modelica, once a value has been given to a variable (most use cases would involve parameters or constants) it is impossible to unset it.  As pointed out in ticket #139, it would be useful to have the ability to "unset" the value or to be able to leave a placeholder value that indicates no value is presently provided.

One way to accomplish this would be to have a built-in variable to symbolically represent an undefined value (e.g. "undefined").  This is present in many languages for both modeling (e.g. MAST) and programming (e.g. Scala, Java, etc).

This would enable code like the following:

```
model A
  parameter Real x = 5.0;
end A;
model B
  extends A(x=undefined);
end B;
```

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/252

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 11 Dec 2012 13:47 UTC**
Update: Note that this ticket is related (but not identical) to #612; and also to issue #923. 

The latter issue is that instead of just making the parameter undefined, we might want to make it undefined in order to allow it to be solved from the initial values:
```
model A
  parameter Real x=5.0;
end A;
model B
  extends A( x(fixed=false)=unbound );
initial equation
  x+1=time;
end B;
```
In this case the intent would not be to force the user to set the parameter x; but only to bring x back to the default state of not having a binding equation. 

How to combine these proposals; and how this relates to balanced models is not fully clear to me at the moment.

**Comment by modelica-trac-importer** [comment]
**Changelog removed by hansolsson on 11 Dec 2012 13:47 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 1 Apr 2016 15:06 UTC**
Merged with #612.

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 1 Apr 2016 16:14 UTC**
